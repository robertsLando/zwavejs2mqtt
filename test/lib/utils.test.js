jest.mock('app-root-path', () => 'foo')

const mod = require('../../lib/utils')

describe('#utils', () => {
  describe('#getPath()', () => {
    test('write && process.pkg', () => {
      process.pkg = true
      return expect(mod.getPath(true)).toBe(process.cwd())
    })
    test('write && !process.pkg', () => {
      process.pkg = false
      return expect(mod.getPath(true)).toMatchSnapshot()
    })
    test('!write && process.pkg', () => {
      process.pkg = true
      return expect(mod.getPath(false)).toMatchSnapshot()
    })
    test('!write && !process.pkg', () => {
      process.pkg = false
      return expect(mod.getPath(false)).toMatchSnapshot()
    })
  })
  describe('#joinPath()', () => {
    test('bool true', () =>
      expect(mod.joinPath(true)).toMatchSnapshot())
    test('bool false', () =>
      expect(mod.joinPath(true)).toMatchSnapshot())
    test('1 length', () =>
      expect(mod.joinPath('foobar')).toMatchSnapshot())
    test('first arg bool gets new path 0', () =>
      expect(mod.joinPath(true, 'bar')).toMatchSnapshot())
  })
})
