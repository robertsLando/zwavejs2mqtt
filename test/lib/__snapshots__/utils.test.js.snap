// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#utils #getPath() !write && !process.pkg 1`] = `"foo"`;

exports[`#utils #getPath() !write && process.pkg 1`] = `"foo"`;

exports[`#utils #getPath() write && !process.pkg 1`] = `"foo"`;

exports[`#utils #joinPath() 1 length 1`] = `"foobar"`;

exports[`#utils #joinPath() bool false 1`] = `"foo"`;

exports[`#utils #joinPath() bool true 1`] = `"foo"`;

exports[`#utils #joinPath() first arg bool gets new path 0 1`] = `"foo/bar"`;
